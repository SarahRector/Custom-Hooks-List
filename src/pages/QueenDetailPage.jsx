import React from 'react';
import QueenDetail from '../components/queen-detail/QueenDetail';

const QueenDetailPage = () => (
  <QueenDetail />
);

export default QueenDetailPage;
