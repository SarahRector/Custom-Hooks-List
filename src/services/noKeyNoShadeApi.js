/* eslint-disable max-len */
export const getQueens = async(page = 1) => {
  const res = await fetch(`http://www.nokeynoshade.party/api/queens/all?page=${page}`);
  const json = await res.json();

  if(!res.ok) throw 'Unable to fetch';

  return json.results.map(queen => ({
    id: queen.id,
    name: queen.name,
    image_url: queen.image
  }));
};

export const getQueensById = async(id) => {
  const res = await fetch(`http://www.nokeynoshade.party/api/queens/${id}`);
  const json = await res.json();

  if(!res.ok) throw 'Unable to fetch';

  return {
    id: json.id,
    name: json.name,
    image_url: json.image,
    quote: json.quote
  };
};
