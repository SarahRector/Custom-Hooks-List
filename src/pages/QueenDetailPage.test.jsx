import React from 'react';
import { render, cleanup, waitFor } from '@testing-library/react';
import QueenDetailPage from './QueenDetailPage';
import { MemoryRouter } from 'react-router-dom';
import { getQueensById } from '../services/noKeyNoShadeApi';

jest.mock('../services/noKeyNoShadeApi.js');

describe('QueenDetailPage component', () => {
  afterEach(() => cleanup());
  it('renders QueenDetailPage', () => {
    getQueensById.mockResolvedValue([
      { id: 89, name: 'Trixie', image_url: 'test.png' }
    ]);
    const { asFragment } = render(<MemoryRouter>
      <QueenDetailPage />
    </MemoryRouter>);
    return waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    }); 
  });
});
