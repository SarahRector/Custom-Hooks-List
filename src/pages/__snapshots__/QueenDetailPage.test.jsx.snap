// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QueenDetailPage component renders QueenDetailPage 1`] = `
<DocumentFragment>
  <h1>
    Loading...
  </h1>
</DocumentFragment>
`;

exports[`QueenDetailPage component renders QueenDetailPage 2`] = `
<DocumentFragment>
  <h1 />
  <img />
  <p />
</DocumentFragment>
`;
